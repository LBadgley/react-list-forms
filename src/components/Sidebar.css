.Sidebar {
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  width: 15vw;
  height: 100vh;
  padding: 20px;
  margin-right: 70px;

  background-color: black;
  color: rgb(255, 163, 188);

  text-align: center;

  a {
    display: block;
    color: rgb(255, 163, 188);
    font-size: 3em;
    text-decoration: none;
  }
}


