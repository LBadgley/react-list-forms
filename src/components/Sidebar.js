import React, { PureComponent } from 'react';

export default class textFormatter extends PureComponent {
  state = {
    text: '',
    color: 'black',
  };
  changeHandler = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }
  
  render() {
    const { text, color, } = this.state;

    return (
      <form>
        <input type="text" name="text" value={text} onChange={this.changeHandler} />
        <input name="color" type="color" value={color} onChange={this.changeHandler} />
        <p style={{ color: color }}>{text}</p>
      </form>
    );
  }
}
