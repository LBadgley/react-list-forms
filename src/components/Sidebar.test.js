import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from './Sidebar';

describe('sidebar component test', () => {
  it('renders the sidebar component', () => {
    const wrapper = shallow(<Sidebar  />);
    expect(wrapper).toMatchSnapshot();
  });
});
