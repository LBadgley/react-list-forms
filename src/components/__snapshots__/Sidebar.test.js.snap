// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar component test renders the sidebar component 1`] = `
<nav
  className="Sidebar"
/>
`;
